packages/manager/modules/pci/src/projects/new/routing.js

```diff
@@ -0,0 +1,193 @@
+import set from 'lodash/set';
+
+import EnvironmentService from '@ovh-ux/manager-config';
+
+import component from './component';
+
+import PciEligibility from './classes/eligibility.class';
+import PciVoucher from './components/voucher/voucher.class';
+import PciPaymentMethodChallenge from './payment/components/challenge/challenge.class';
+
+import {
+  ELIGIBILITY_ACTION_ENUM,
+  ELIGIBILITY_ERROR_IMAGES_SRC,
+} from './constants';
+
+import { PCI_REDIRECT_URLS } from '../../constants';
+
+export default /* @ngInject */ ($stateProvider, ovhFeatureFlippingProvider) => {
+  if (ovhFeatureFlippingProvider.isFeatureActive('pci.onboarding.new')) {
+    $stateProvider.state('pci.projects.new', {
+      url: '/new?cartId&voucher',
+      redirectTo: (transition) => {
+        const translatePromise = transition.injector().getAsync('$translate');
+        const windowPromise = transition.injector().getAsync('$window');
+        const cartPromise = transition.injector().getAsync('cart');
+        const eligibilityPromise = transition
+          .injector()
+          .getAsync('eligibility');
+        return Promise.all([
+          translatePromise,
+          windowPromise,
+          cartPromise,
+          eligibilityPromise,
+        ]).then(([$translate, $window, cart, eligibility]) => {
+          let redirectState = 'pci.projects.new.config';
+          let redirectParams = transition.params();
+          const redirectOptions = {
+            location: false,
+          };
+
+          if (eligibility.isAskIncreaseProjectsQuotaRequired()) {
+            redirectState = 'pci.error';
+            redirectParams = {
+              message: $translate.instant(
+                'pci_project_new_error_ask_increase_projects_quota',
+              ),
+              code: ELIGIBILITY_ACTION_ENUM.ASK_INCREASE_PROJECTS_QUOTA,
+              image: ELIGIBILITY_ERROR_IMAGES_SRC.ASK_INCREASE_PROJECTS_QUOTA,
+              submitLabel: $translate.instant(
+                'pci_project_new_error_contact_support',
+              ),
+              submitLink: get(
+                PCI_REDIRECT_URLS,
+                `${EnvironmentService.Environment.region}.support`,
+              ),
+            };
+          } else if (eligibility.isVerifyPaypalRequired()) {
+            redirectState = 'pci.error';
+            redirectParams = {
+              message: $translate.instant(
+                'pci_project_new_error_verify_paypal',
+                {
+                  href: get(
+                    PCI_REDIRECT_URLS,
+                    `${EnvironmentService.Environment.region}.paymentMethods`,
+                  ),
+                },
+              ),
+              code: ELIGIBILITY_ACTION_ENUM.VERIFY_PAYPAL,
+              image: ELIGIBILITY_ERROR_IMAGES_SRC.VERIFY_PAYPAL,
+              submitLabel: null,
+            };
+          } else if (cart.cartId !== transition.params().cartId) {
+            $window.location.replace(
```

Comment:
Why not just return the wanted state ?

Reply:
Because it doesn't work as expected. The cartId param is not set to the URL if I redirect to the state. The only way I found to have it in the URL, is to replace the location and then reload it... :-(